### PR TITLE
feat(combobox-item): add `heading` property (deprecates `textLabel`) and add `label` property

### PR DESCRIPTION
**Pull request by aPreciado88**
**Related Issue:** [#8737](https://github.com/Esri/calcite-design-system/issues/8737)

## Summary
Update `combobox` component to:

- Deprecate `textLabel ` in favor of `heading`.
- Add `heading` prop, which will fallback to `textLabel` if unspecified.
- Add `label` prop, which will fallback to `value` if not specified.

### CODE REVIEW

File: packages/calcite-components/src/components/combobox-item/combobox-item.tsx [hunk 1]
```diff
@@ -136,11 +136,21 @@ export class ComboboxItem implements ConditionalSlotComponent, InteractiveCompon
   @Prop() shortHeading: string;
 
   /** The component's text. */
+  @Prop({ reflect: true }) heading: string;
```
Comment (jcfranco) [inline packages/calcite-components/src/components/combobox-item/combobox-item.tsx:139]: Let's not reflect `heading`, as it is content. I don't recall why we were originally reflecting `textLabel`. 🤔

**Review by jcfranco — APPROVED** [review]
Looks great! 

Before merging, could you revisit the PR title before merging to highlight the new props? This is important as the titles are automatically pulled into the changelog. It’s helpful to frame them from a developer’s perspective, focusing on the benefits or changes they’ll see (similar to [this PR](https://github.com/Esri/calcite-design-system/pull/9194)).

**Comment by aPreciado88** [comment]
@jcfranco I updated the PR with the requested changes. Does the new title look okay? 🐱‍👤

File: packages/calcite-components/src/components/combobox-item/combobox-item.tsx [hunk 2]
```diff
@@ -136,11 +136,21 @@ export class ComboboxItem implements ConditionalSlotComponent, InteractiveCompon
...
 
   /** The component's text. */
+  @Prop() heading: string;
+
+  /**
+   * The component's text.
+   *
+   * @deprecated Use `heading` instead.
+   */
   @Prop({ reflect: true }) textLabel!: string;
```
Comment (driskull) [inline packages/calcite-components/src/components/combobox-item/combobox-item.tsx:146]: `textLabel` should remove the `!` so its no longer required here.

**Pull request merged**